plugins/org.obeonetwork.dsl.uml2.design/src/org/obeonetwork/dsl/uml2/design/api/services/ClassDiagramServices.java

```diff
@@ -545,10 +932,12 @@ public boolean sourceIsComposite(Association association) {
 	 *
 	 * @param association
 	 *            Association
+	 * @param element
+	 *            Edge element
 	 * @return True if source is navigable
 	 */
-	public boolean sourceIsNavigable(Association association) {
-		final Property source = AssociationServices.INSTANCE.getSource(association);
+	public boolean sourceIsNavigable(Association association, DDiagramElement element) {
+		final Property source = getSourceEndNaryAssociation(association, element);
```

Comment:
if the association is not an N-ary one we still go through the sourceIsNavigable method so the name of the getSourceEndNAryAssociation is weird